setup.py

```diff
@@ -50,40 +44,43 @@ def build_extension(self, ext):
                 extra_link_args.append("-Wl,-so%s" % file_name)
             else:
                 extra_link_args.append("-Wl,-soname,%s" % file_name)
-        distutils.command.build_ext.build_ext.build_extension(self, ext)
+        super().build_extension(ext)
 
     def finalize_options(self):
-        distutils.command.build_ext.build_ext.finalize_options(self)
+        super().finalize_options()
         if sys.platform == "win32" and self.build_implib is None:
-            target_dir = "implib.%s-%s" % \
-                    (distutils.util.get_platform(), sys.version[:3])
-            self.build_implib = os.path.join("build", target_dir)
+            build_dir = os.path.dirname(self.build_lib)
+            target_dir = "imp" + os.path.basename(self.build_lib)
+            self.build_implib = os.path.join(build_dir, target_dir)
 
     def initialize_options(self):
-        distutils.command.build_ext.build_ext.initialize_options(self)
+        super().initialize_options()
         if sys.platform == "win32":
             self.build_implib = None
 
 
 # define class to ensure that the import library and include file is installed
 # properly; this is relevant on Windows platform only.
-class install_data(distutils.command.install_data.install_data):
+class install(setuptools.command.install.install):
 
     def run(self):
-        distutils.command.install_data.install_data.run(self)
+        super().run()
         if sys.platform == "win32":
+            # library (.lib or .a)
             command = self.get_finalized_command("build_ext")
-            target_dir = os.path.join(self.install_dir, "Libs")
+            if command.compiler.compiler_type == "msvc":
```

Comment:
These changes have nothing to do with switching to setuptools (which I support). Technically they should be in a separate PR explaining their purpose.

Reply:
They work differently in msvc and mingw.

Reply:
Sure. Sounds like that should be a separate PR with an explanation. Can you do that? I presume you have tested these changes on Windows yourself?

Reply:
I already reverted. I'll make a separate PR for mingw.

Reply:
Tested on windows and linux. This part on windows only.

Reply:
Great. Thanks.